src/queries/useAuthoritiesBrowse/useAuthoritiesBrowse.js

```diff
@@ -203,12 +203,8 @@ const useAuthoritiesBrowse = ({
     setItems(mainRequest.data?.items || []);
   }, [mainRequest.data]);
 
-  useEffect(() => {
-    if (page === 0) {
-      const dataIncludesEmptyAnchor = !!mainRequest.data?.items.find(item => !item.authority);
-
-      setHasEmptyAnchor(dataIncludesEmptyAnchor);
-    }
+  const hasEmptyAnchor = useMemo(() => {
+    return page === 0 && totalRecords !== 0 && !!mainRequest.data?.items.find(item => !item.authority);
```

Comment:
Here in `hasEmptyAnchor` you use `totalRecords` but the thing is that `totalRecords` is defined later in code and moreover it in its turn uses `hasEmptyAnchor`...
Please, figure out the dependency here.